### DISCUSSION TITLE
Error no module named 'network'

### DISCUSSION THREAD

**Discussion by jazzjohn27** [micropython/micropython#10069] [Using MicroPython]
A few months ago I wrote code for the Pico W and successfully imported the network module. Revisiting the same code n ow  results in an  error importing the network module. 
What changed?

**Comment by jimmo** [comment]
Can you provide some more information?

Have you changed anything on your device? Is it running Pico W (not Pico) firmware?

If you hit Ctrl-B at the REPL, what's the version information that is printed out?

**Comment by peterhinch** [comment]
I hit this the other day. The answer was embarrassingly simple: I'd installed firmware for Pico instead of Pico W.

**Comment by jazzjohn27** [comment]
I might be confusing 2 issues here. Let me start simply...
Running Win 10. Plugged in a new Pico W with reset button pressed to mount as a drive.
Downloaded micropython-firmware-pico-w-290622.uf2 and copied it to the Pico W.
After that unplugged and plugged back into the USB on the computer.
In Thonny, I am now able to import network.

On a third Pico W (not the one running my code) I went through the above steps and it keeps connecting and disconnecting as a serial port ( once every few seconds.)

However, the problem is with  the PIco W that has the main.py code (that has been running stand-alone for a few months.)
After plugging it in and configuring the port and board in Thonny,
I get this in Thonny's shell window:

Traceback (most recent call last):
  File "main.py", line 189, in <module>
  File "uasyncio/core.py", line 1, in run
  File "uasyncio/core.py", line 1, in run_until_complete
  File "uasyncio/core.py", line 1, in wait_io_event
KeyboardInterrupt: 
MicroPython v1.19.1 on 2022-10-14; Raspberry Pi Pico W with RP2040
Type "help()" for more information.
MicroPython v1.19.1 on 2022-10-14; Raspberry Pi Pico W with RP2040

Type "help()" for more information.

>>> 
Connection lost -- EOF

Use Stop/Restart to reconnect.

Process ended with exit code 1.


Needless to say, without a connection  I cannot get at the main.py file on the Pico W.
I have changed nothing on this Pico W since I first wrote the code and put it into service a few months ago. Could something change on the device by just being powered on and running main.py?

**Reply by robert-hh** [reply]
Do you get the same sequence when NOT using Thonny with all it's magic under the hood, but a dump terminal emulator like Putty or TeraTerm?

**Reply by jazzjohn27** [reply]
That was my next question.  I tried Puttying in to the com port, but that does nothing. Is there a way to directly edit the .py files on a Pico W that does not run network code? Thonny does seem to introduce an extra layer.
When the Pico is mounted as a USB drive, it would be nice to be able to see the files, but apparently they are hidden.

**Reply by robert-hh** [reply]
With Putty you have to disable flow control. You can as well use mpremote.py with it's subcommands.

`mpremote --help`

shows as short help page.

`mpremote connect COMx`

gets you in a REPL terminal.

`mpremote connect COMX ls`

shows the files.

`mpremote connect COMx edit filename`

Allows editing a file on the device. Unless you just open a REPL terminal, mpremote attempts to stop a running script.

**Reply by jazzjohn27** [reply]
mpremote connect COM6 ls        does list the main.py file... 
but
mpremote connect COM6 edit filename       generates error "edit: $EDITOR not set"
How can I get the editor to work?

**Reply by jimmo** [reply]
> "edit: $EDITOR not set"

Sorry this is not currently supported on Windows. See https://github.com/micropython/micropython/pull/9609 (I'll try and get that merged ASAP).

In the meantime you can do this manually:

```
mpremote connect COMx cp :main.py main.py
# edit main.py
mpremote connect COMx cp main.py :main.py
```
